packages/python/plotly/plotly/express/_core.py

```diff
@@ -929,9 +929,9 @@ def build_dataframe(args, attrables, array_attrables):
                             keep_name = df_provided and argument is df_input.index
                         else:
                             # we use getattr/hasattr because of index
-                            keep_name = hasattr(
-                                df_input, col_name
-                            ) and argument is getattr(df_input, col_name)
+                            keep_name = col_name in df_input and argument.equals(
```

Comment:
Is it intentional to move from `is` to `.equals()` here?

Reply:
(also we should remove the comment above now ;)

Reply:
yes it was intentional, I think it's enough that the values are all equal. What do you think?

Reply:
ok I reverted to `is`